src/java.desktop/windows/native/libawt/windows/ShellFolder2.cpp

```diff
@@ -912,15 +912,51 @@ JNIEXPORT jint JNICALL Java_sun_awt_shell_Win32ShellFolder2_getIconIndex
     return (jint)index;
 }
 
+/*
+ * Class:     sun.awt.shell.Win32ShellFolder2
+ * Method:    hiResIconAvailable
+ * Signature: (JJ)Z
+ */
+JNIEXPORT jboolean JNICALL Java_sun_awt_shell_Win32ShellFolder2_hiResIconAvailable
+    (JNIEnv* env, jclass cls, jlong pIShellFolderL, jlong relativePIDL)
+{
+    IShellFolder* pIShellFolder = (IShellFolder*)pIShellFolderL;
+    LPITEMIDLIST pidl = (LPITEMIDLIST)relativePIDL;
+    if (pIShellFolder == NULL || pidl == NULL) {
+        return FALSE;
+    }
+    HRESULT hres;
+    IExtractIconW* pIcon;
```

Comment:
`pIcon->Release()` must be called before returning as done in `extractIcon`.

Reply:
Fixed.